calico_node/ip_detection.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strings"
+)
+
+func detectIPs(ignoreInterfaces []string) ([]string, error) {
+
+	ips := []string{}
+	ifaces, err := net.Interfaces()
+
+	for _, iface := range ifaces {
+
+		valid_ip := true
+
+		// Check each interface name against 
+		// list of interfaces to ignore.
+		for _, v := range ignoreInterfaces {
+			regex, err := regexp.Compile(v)
+
+			if err != nil {
+				continue
+			}
+
+			if regex.MatchString(iface.Name) {
+				valid_ip = false
+			}
+		}
+
+		if valid_ip == true {
+			addrs, err := iface.Addrs()
+
```

Comment:
Same here - I'd remove the extra newline so it's clear which error we're handling.